File: src/main/java/com/intellij/tapestry/intellij/view/nodes/AbstractModuleNode.java
package com.intellij.tapestry.intellij.view.nodes;

import com.intellij.ide.projectView.PresentationData;
import com.intellij.openapi.module.Module;
import com.intellij.ui.treeStructure.SimpleNode;
import com.intellij.util.PlatformIcons;
import org.jetbrains.annotations.NotNull;

public abstract class AbstractModuleNode extends TapestryNode {

    public AbstractModuleNode(@NotNull final Module module) {
        super(module);

        init(module, new PresentationData(module.getName(), module.getName(), PlatformIcons.WEB_ICON, null));
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public abstract SimpleNode @NotNull [] getChildren();
}

File: src/main/kotlin/com/github/rar91279/plugin/tapestry/intellij/view/nodes/AbstractModuleNode.kt
package com.github.rar91279.plugin.tapestry.intellij.view.nodes

import com.intellij.ide.projectView.PresentationData
import com.intellij.openapi.module.Module
import com.intellij.ui.treeStructure.SimpleNode
import com.intellij.util.PlatformIcons

abstract class AbstractModuleNode(module: Module) : TapestryNode(module) {

    init {
        init(module, PresentationData(module.name, module.name, PlatformIcons.WEB_ICON, null))
    }

    abstract override fun getChildren(): Array<SimpleNode>
}
